static/js/components/Navigation.js

```diff
@@ -14,25 +16,33 @@ import { userIsAnonymous } from "../lib/util"
 import type { Channel } from "../flow/discussionTypes"
 
 type NavigationProps = {
+  channels: Map<string, Channel>,
   pathname: string,
   subscribedChannels: Array<Channel>
 }
 
 const Navigation = (props: NavigationProps) => {
-  const { subscribedChannels, pathname } = props
+  const { channels, subscribedChannels, pathname } = props
 
   const channelName = getChannelNameFromPathname(pathname)
 
+  const currentChannel = channels.get(channelName)
+  const showPostButton =
+    !userIsAnonymous() &&
+    (currentChannel
+      ? userCanPost(currentChannel)
+      : R.any(channel => userCanPost(channel), Array.from(channels.values())))
```

Comment:
can this just be `R.any(userCanPost)`? Also I think you can use the spread operator with `.values()`, up to you (like `[...channels.values()]`)